docs/product/components/popovers.html

```
@@ -261,6 +267,99 @@ <h3 class="fs-body2 mb2">Teams</h3>
+                <button class="s-btn s-btn__link" data-action="section#showHelp">Click here for help</button>
+            </div>
+        </div>
+    </div>
+    <script>
+        window.addEventListener("load", () => {
+                Stacks.addController("section", {
+                targets: ["help"],
+                showHelp(e) { Stacks.showPopover(this.helpTarget); e.stopPropagation(); },
+                hideHelp(e) { Stacks.hidePopover(this.helpTarget); },
+            });
+        });
+    </script>
+
+    {% header "h4", "JavaScript configuration" %}
+    <p class="stacks-copy">Situations may also arise where popovers need to be attached to an element after the document is rendered.  For example, a button could have a contextual menu that is too expensive to serve up on every page load.</p>
+    <p>Popovers can be attached to an element after the fact using <code class="stacks-code">Stacks.attachPopover</code>.</p>
+
+    <p>This method takes three parameters, the element to attach the popover to, the popover either as an element or an HTML string, and optional options for displaying the popover.</p>
+
+    <div class="stacks-preview">
+{% highlight javascript %}
+Stacks.application.register("actions", class extends Stacks.StacksController {
```

I'd write this type of scenario a bit more simply:

```html
<button id="test1">Show popover</button>
<div id="container1"></div>
<script>
document.querySelector('#test1').addEventListener('click', function(e){
    var popoverElement = document.querySelector('#container1 [data-controller="s-popover"]');

    if (popoverElement) {
        Stacks.showPopover(...);
    }
    else {
        //TODO fetch the popover, append to container1 (letting stimulus do the hookup on mutation) and then calling show
    }
});
</script>
```

---

That's a huge `//TODO` right there.  In all my scenarios, the button you'd click is the same as the s-popover's reference element.  Per the current implementation, it can't have `data=controller="s-popover"` until the popover is loaded or it will throw errors.  That means we need code to add the controller to the button after it loads.

---

Ah, we can't have the popover missing at validation time, right. Hmm... We'd definitely want to support this case natively without hacks, but if we don't have a named popover target, we attempt to generate it, which will fail for the base popover. I'd love to suppress that error in this case and just let the consumer keep writing the controller like they're used to.

An alternative would be to have a "loading"/placeholder popover as a dummy to bypass validation, then load the real one in on click later.

Maybe we should add `s-popover-async` and `s-modal-async` (loadModal replacement) controllers for something like this. Seems common enough.

Let's punt on this for now and revisit soon. Resolving.